sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -519,6 +521,7 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         case e if !e.resolved => u
         case g: Generator => MultiAlias(g, Nil)
         case c @ Cast(ne: NamedExpression, _, _, _) => Alias(c, ne.name)()
+        case se @ ScopedExpression(ne: NamedExpression, _) => Alias(se, ne.name)()
```

Comment:
just in case, shall we also match `ScopedExpression(ExtractValue...)` and `ScopedExpression(Literal...)`?

Reply:
What about
```suggestion
        case se @ ScopedExpression(e: Expression, _) =>
          resolve(UnresolvedAlias(e, optGenAliasFunc)) match {
            case ne: NamedExpression => Alias(se, ne.name)()
            case _ => se
          }
```

Reply:
LGTM

Reply:
Fixed in 9a360dd6465.